src/modules/game_func.py

```diff
@@ -0,0 +1,78 @@
+import os
+import sys
+import cv2 as cv
+import math
+from modules.hand_tracking import HandDetector
+from modules.piano_key import PianoKey
+from modules.piano import Piano
+sys.path.append("..")
+
+class Game:
+    detector = None
+    monitor = None
+    success, img = None, None
+    height, width = None, None
+    piano = None
+    spath = None
+    turn = None
+    cond = 15
+    pianolen = None
+    indent = None
+    pressed = None
+
+    def __init__(self, cap, path, turn=1, octave=3, key_num=14):
+        self.turn = turn
+        self.detector = HandDetector()
+        self.success, self.img = cap.read()
+        self.height, self.width = self.img.shape[:2]
+        self.piano = Piano(int(self.width / 50), int(self.height / 50),
+                      self.width, int(self.height / 2))
+        self.spath = path
+        self.piano.key_generator(self.spath, octave, key_num)
+        self.pianolen = len(self.piano.keys)
+        self.indent = int(self.width / 50)
+        self.pressed = {}
+        for key in self.piano.keys:
+            self.pressed[key] = False
+
+    def balance(self, id):
+        if id == 4:
+            return 6
+        elif id == 20:
+            return -2
+        return 0
+
+    def render(self, img):
+        img = cv.flip(img, self.turn)
+        left_points, right_points = self.detector.findPosition(img, True)
+        fingers = []
+        zone = self.piano.get_key_shape(0)[0]

```

Comment:
`AttributeError: 'Piano' object has no attribute 'get_key_shape'`

update your PR to main pls